tests/AbstractLexerTest.php

```diff
@@ -279,4 +282,60 @@ public function testMarkerAnnotationLocaleTr(): void
         self::assertEquals('@', $mutableLexer->token->value);
         self::assertEquals('ODM\Id', $mutableLexer->lookahead->value);
     }
+
+    public function testCanTokenizeFloatValue(): void
+    {
+        $lexer = new /** @template-extends AbstractLexer<int, string|int|float|bool> */ class () extends AbstractLexer {
+            final public const T_NONE    = 1;
+            final public const T_INTEGER = 2;
+            final public const T_FLOAT   = 4;
+            final public const T_BOOL    = 4;
```

Comment:
Why do those constants have the same value?

Reply:
This does not have any consequences, but I fixed it anyway.